### DISCUSSION TITLE
based CID decentralized IP address, Abolish ipv4 and ipv6

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [ipfs/ipfs#510] [Poll]
THE IDEAL PATH
What if we replace the controlled and surveilled IP addresses with new internet addresses that are based on CID? These addresses will inherit all CID’s features, i.e., they will be purely decentralized, secure, future-proof, robust, anonymous, unhackable, controlled by no single authority and many more.

Is it just a dream? For now. If this could be true we would be changing the internet as we know it.

* CID address + CID DNS, access to decentralized domain

win10 network adapters : Enter the CID address
Wi-Fi  DHCP Static : Enter the CID address

---
IPv4 and IPv6 are centralized. They are controlled by national companies, etc. 
![](https://www.apnic.net/wp-content/uploads/policy-environment_obsolete/images/ipv4_hierarchy.png)

Try to Google “Who controls IP addresses?” You’ll promptly get “IANA: the Internet Assigned Numbers Authority.” IANA is the top authority behind IP address allocation and assignment. There are five different regional internet registries (RIR) with jurisdiction under the IANA.

As a matter of fact, as an individual or a normal internet user you cannot request IP addresses directly from IANA or one of the five RIRs, but only from internet service providers, such as the services offered by mobile or telecom operators.

* True decentralization, no official singular Foundation, Committee, Corporation, or entities in permanent unitary control of the protocol.

**Comment by Talleyrand-34** [comment]
The main problem is routing, if you are not assigned any ip adress, how is your package routed? You will be isolated from anyone else. If you want to use CID you need to build over ip infrastructure

**Comment by ghost** [comment]
Creating a decentralized internet addressing system based on Content Identifier (CID) addresses rather than traditional IP addresses could have several features:

1. **Global Scalability**: CID addresses can potentially provide a more scalable solution compared to IPv4 and IPv6, allowing for a virtually unlimited number of unique addresses.

2. **Content-Centric**: CID addresses are content-centric, meaning they are tied to the content itself rather than the location or network it resides on. This aligns well with the decentralized nature of the internet and could facilitate content distribution and retrieval.

3. **Improved Security**: With a decentralized addressing system, there may be opportunities to enhance security and privacy, potentially mitigating issues like IP address spoofing and DDoS attacks.

4. **Peer-to-Peer Communication**: CID addresses could enable more efficient peer-to-peer communication, reducing reliance on centralized servers for content delivery and facilitating direct communication between users.

5. **Reduced Dependency on ISPs**: By moving away from traditional IP addressing, there could be less reliance on Internet Service Providers (ISPs) for assigning and managing IP addresses, potentially leading to greater independence for users.

6. **Interoperability**: Any new addressing system would need to ensure compatibility and interoperability with existing internet protocols and infrastructure to facilitate a smooth transition.

**Reply by Talleyrand-34** [reply]
Yes, but that means replacing ipv4/ipv6 or at least ISPs need to support a new layer 3 protocol. On one hand you have to build a new layer 3 protocol and on the other build over IP. I do not discuss the benefits, i discuss the implementation, and the main problem in implementation is public routing where you depend on ISPs.
